code/datums/components/customizable_reagent_holder.dm

```diff
@@ -289,3 +291,9 @@
 	context[SCREENTIP_CONTEXT_LMB] = "[screentip_verb] [held_item]"
 
 	return CONTEXTUAL_SCREENTIP_SET
+
+/// Clear refs if our food goes away
+/datum/component/customizable_reagent_holder/proc/food_exited(datum/source, atom/movable/gone)
+	SIGNAL_HANDLER
+	if(gone in ingredients)
+		LAZYREMOVE(ingredients, gone)
```

Comment:
why the in check, if it's not it'll do nothing (in the same time cause they're both O(n)

Reply:
oh yeah